### DISCUSSION TITLE
yarn analyze-function-bundle command

### DISCUSSION THREAD

**Discussion by ThomasAribart** [floydspace/serverless-esbuild#219] [Ideas]
Hello @floydspace and thanks for this great lib !

I've created a small script that uses `serverless-esbuild` in combination with [`esbuild-visualizer`](https://www.npmjs.com/package/esbuild-visualizer) to plot the bundle of a lambda. What it looks like:

https://user-images.githubusercontent.com/38014240/139452311-20357329-9af8-4f62-a986-cd53c784136d.mp4

Would you be interested in this script becoming a feature ? Something like:

```
yarn sls visualize-esbuild-bundle myFunctionName [...slsOptions]
```

Cheers,

Thomas

**Comment by adriencaccia** [comment]
Hey !

I really like this idea.

In the meantime, I created https://github.com/adriencaccia/serverless-analyze-bundle-plugin to make it available as a plugin.

**Reply by floydspace** [reply]
@adriencaccia  Am I right that the plugin is working only together with `serverless-esbuild`?

Would be nice if you'd contribute in the readme that this plugin exists.

**Reply by adriencaccia** [reply]
Yes indeed, it works only with `serverless-esbuild` !

I created #220 quickly, let me know you if it's okay for you or if you imagined something different :slightly_smiling_face:

**Comment by olup** [comment]
I think it makes perfect sense as a plugin ! Thanks for the script and the
plugin !

**Comment by floydspace** [comment]
@ThomasAribart thanks for the idea.
I agree with @olup, I'd prefer having a composition of modules instead of a swiss knife when it's possible.

**Comment by floydspace** [comment]
@all-contributors
please add @ThomasAribart for ideas.
